ui/components/download-button.vue

```diff
@@ -209,27 +304,131 @@ export default {
     };
   },
   computed: {
-    isSingleFixture() {
-      return typeof this.download === `string`;
+    // returns, whether we're handling only one single fixture here
+    // or all fixtures in a specific format
+    isSingle() {
+      if (this.editorFixtures && this.editorFixtures.fixtures.length === 1) {
+        return true;
+      }
+
+      if (this.fixtureKey) {
+        return true;
+      }
+
+      return false;
     },
     title() {
-      if (this.isSingleFixture) {
-        return `Download as …`;
+      if (this.isSingle) {
+        return `Download as…`;
       }
 
-      return `Download all ${this.download} fixtures`;
+      return `Download all ${this.fixtureCount} fixtures`;
     },
     baseLink() {
-      if (this.isSingleFixture) {
-        return `/${this.download}`;
+      if (this.editorFixtures) {
+        return `/download-editor`;
+      }
+
+      if (this.isSingle) {
+        return `/${this.fixtureKey}`;
       }
 
       return `/download`;
+    },
+    isStyleHome() {
+      return this.buttonStyle === `home`;
+    },
+    isStyleBig() {
+      return this.buttonStyle === `big` || this.isStyleHome;
     }
   },
   methods: {
-    blur(event) {
+    downloadDataAsFile(blob, filename, type) {
+      if (typeof window.navigator.msSaveBlob !== `undefined`) {
+        // IE workaround for "HTML7007: One or more blob URLs were revoked by closing the blob for which they were created.
+        // These URLs will no longer resolve as the data backing the URL has been freed."
+        window.navigator.msSaveBlob(blob, filename);
+      }
+      else {
+        const URL = window.URL || window.webkitURL;
+        const downloadUrl = URL.createObjectURL(blob);
+
+        const anchorElement = document.createElement(`a`);
+
+        if (typeof anchorElement.download === `undefined`) {
+          window.location = downloadUrl;
+        }
+        else {
+          anchorElement.href = downloadUrl;
+          anchorElement.download = filename || ``;
```

Comment:
`filename` is actually always a string. Even if not, could we use parameter defaults in this function?